src/main/java/net/minecraftforge/fml/common/versioning/Restriction.java

```diff
@@ -212,9 +212,9 @@ public String toStringFriendly()
         }
         else if ( getLowerBound() != null && getUpperBound() != null )
         {
-            if ( getLowerBound().equals(getUpperBound()) )
+            if ( getLowerBound().getVersionString().equals(getUpperBound().getVersionString()) )
             {
-                return getLowerBound().toString();
+                return getLowerBound().getVersionString();
```

Comment:
Can someone confirm if these changes are ok (it prevents returning something like beetween 1.10.2 and 1.10.2)